Capstone Project

VelzckC0D3 opened the pull request:
# 👨‍🎓 CAPSTONE PROJECT

## ♻️ _Requested changes PR 1_  ♻️
✔️ _Corrected the `<hx>` hierarchy so it haves correct semantics _
✔️ _Now, the about page and the home page are 2 separate documents_
✔️ _Corrected a visual bug on the hamburger that was shown whenever the solution was changed and the menu was open_

## ✨ _Requested Implementations_ ✨
✔️ _Personalized the content of the page, the topic was "Architecture & Design"_
✔️ _Followed palette of colors, typos, size, and weight, using the one provided `Lato`_
✔️ _Only changes images and text but the structure is the same_
✔️ _Base CSS_
✔️ _Included Home page & About page_
✔️ _2 different screen sizes: `370`  `768` also added tablets and smaller phone screens_

### ✨ _Interactions_ ✨
✔️ _The home page has the link to the about page on the desktop version, and on the menu for the mobile version_
✔️ _The logo in the header at desktop & tablet versions takes the user to the home page_
✔️ _When the user tabs the button for the menu it is displayed over the page_
✔️ _Estilized mobile menu adapted to my theme_
✔️ _The mobile menu have an X button to close the menu_

### ✨ _Dynamic Page_ ✨
✔️ _The section Featured Speakers is created dynamically using Javascript,  `also, I added functionality to the "more" button_
✔️ _The website dynamically generates the HTML card div taking the information from the speaker object to display it_

## ✨ _Preview & Live Demo_ ✨
### [_**Here, you can check the live demo of the actual version**_](https://velzck-festival.netlify.app/) 
![showcase-website](https://user-images.githubusercontent.com/92229666/226081133-ec20545e-2041-410d-a86b-1a4d1d6c28d5.gif)

## ✨ _Extra Features_  ✨
✔️ _Smooth animations and adaptative boxes `flexBox & cssGridLayout`._
✔️ _Included a feature that allows the user to press the key `ESC` to close the mobile menu and the pop-up_

### **_I want to say thanks to Cindy Shin [Cindy Shin](https://www.behance.net/adagio07) who allowed the usage of her template_ 🫡**

# ✨ _Loom Video_ ✨
## [_Click here to go to the video_](https://www.loom.com/share/c7e2bddaa78245be86f94f56c6d8509f)

#  💡 _Thank you for your review!_ 💡 

![ThumbsUpGIF](https://user-images.githubusercontent.com/92229666/226083473-6f2f73c2-9a57-4ea0-809c-3c16f368f572.gif)

VelzckC0D3 committed:
HTML & CSS

- added home page
- added its styles and animations

VelzckC0D3 committed:
Major Update

- started to construct responsiveness
- implemented menu interactions javascript
- implemented dynamic menu javascript
- implemented abou page
- corrected menu

VelzckC0D3 committed:
Finished Project

- Fully responsive and adaptative design
- function 'see more' on the speakers card
- function when scrolling on desktop version, it puts the header on -fix position
- the about and home page are intercalated with a function that I called switch. This allows a better performance since it doesn't load any content and just executes a script
- function to the accessibility bar that opens the language selector (none of those languages are included)

VelzckC0D3 committed:
Final Update

- double checked all Linters
- checked for possible bugs
- added a function that disables the mobile menu after certain resolution
- corrected minor css bugs
- added 'alt' to all images
- added type button for all buttons

VelzckC0D3 committed:
header tag correction

- corrected header tag

VelzckC0D3 committed:
buttonsActions update

- added function that sends the user to the home page whenever they click the logo

VelzckC0D3 committed:
img correction

- corrected the format for an image on the about page

VelzckC0D3 committed:
corrected some visuals

- final corrections

VelzckC0D3 committed:
checked linters

- corrected eslint linter

VelzckC0D3 committed:
function correction

- fixed a function that didn't worked

VelzckC0D3 committed:
delete unused code

- deleted unused piece of code

VelzckC0D3 committed:
corrected more function

- fixed the 'more' button function at the speakers section in the homepage

VelzckC0D3 committed:
deleted not used script

VelzckC0D3 committed:
font corrections

- fixed grid visual errors
- corrected fonts

VelzckC0D3 committed:
linter correction

- final commit

VelzckC0D3 committed:
final details

- added x button for the menu
- double checked linters

VelzckC0D3 committed:
bug correction

- corrected a bug on the more button

VelzckC0D3 committed:
readme

- added all the readme to my capstone project

shella12 commented on index.html:
```diff
@@ -1,14 +1,254 @@
+    <!-- HERO -->
+    <section id="heroCont">
+
+      <div id="hero">
+        <h3 class="heroTitle">"hello designers of the world!"</h3>
+        <h1 class="heroTitle">architecture and designers festival <br>2023</h1>
+        <p id="heroDescription">Welcome to the intrnational <span id="heroSpan">Architecture and Design festival
+            2023®</span> The best architects and designers around the globe to take the international prestigious
```
- [ ]  Your code is well structured 🏛️ and clean ✨ . Great work 👍 . However It would be better if you use `<hx>` tag hierarchy properly, in a way that `<h1>` tag should come first then `<h2>`,  `<h3>` and so on ...

You can also use `<p>` tag for "hello designers of the world!" text since its just a plain text and not a title or subject to be wrapped in a `heading` tag

The purpose of the `heading` tags is semantics. Correct use of headings on a page provides hierarchy and overview, and helps quickly communicate what a web page is about. They help visitors quickly scan a page, and they help search engines quickly understand a page's structure and what it's about. If you want your heading to be a particular font please use `CSS` properties for that.

VelzckC0D3 replied:
@shella12 Thank you! I've corrected that error, I'll have it in mind from now!
![image](https://user-images.githubusercontent.com/92229666/226091165-7d46fd92-944b-4c11-a4a9-6043c634ece6.png)
![image](https://user-images.githubusercontent.com/92229666/226091180-25ffa08f-61fe-4522-be74-009018737d2e.png)

VelzckC0D3 replied:
@shella12 Thank you for your feedback!, from now on I'll will have it in mind 👌
![image](https://user-images.githubusercontent.com/92229666/226091743-8de74101-a912-43cd-83bf-b6160a8e417f.png)
![image](https://user-images.githubusercontent.com/92229666/226091753-986311f4-f2e7-442c-96e8-1d93d4812d9e.png)

shella12 commented on index.html:
```diff
@@ -1,14 +1,254 @@
+  <!--   HOMEPAGE -->
+
+  <!-- ABOUTPAGE -->
+  <section id="aboutPage" class="switch">
+
+    <section id="heroContAbout">
+
+      <div id="heroAbout">
+        <h3 class="heroTItleAbout">"hello designers of the world!"</h3>
+        <h1 class="heroTItleAbout">architecture and designers festival <br>2023</h1>
+        <p id="aboutDescription">The Architects and Designers Festival brings together a community of experts,
+          academics, and activists in the field of architecture and design in a different location every two years.
+          This year, we aim to expand our guest list to include organizations and individuals interested in
+          collaborating with us on joint projects that promote the advancement of the industry, free expression, and
+          open knowledge. If you're actively involved in the world of architecture and design - advocates for
+          open-source software, design enthusiasts, professionals from galleries, libraries, museums, archives,
+          government bodies, foundations, lawyers, and activists - we invite you to join us this year to work together
+          in creating a more robust and dynamic community.</p>
+        <p id="aboutContact">
+          Please contact us per Email for nay further questions about CC Global Summit 2015! <a id="aboutEmail"
+            href="mailto:alejandro.velzck@gmail.com">alejandro.velzck@gmail.com</a>
+        </p>
+      </div>
+
+    </section>
```
- [ ] Considering the efficiency of the site you thought of using the javascript function instead of creating separate page for about page, excellent thinking 🧠  you are already thinking ahead. However in this capstone we are required to make a separate page for the about section. So I would suggest making a separate page for the` about page` instead of making a section in the index page. The purpose of this activity is learning and to stick to the client requirements as sometimes the client asks for lemon and you provide them with lemonade 🍹 but the client just wants a lemon 🍋 . 

![client requirements](https://user-images.githubusercontent.com/44798044/226085541-42656535-c52e-446a-b7cc-7143ddfab7a3.png)

VelzckC0D3 replied:
Thank you for your feedback I've already implemented the way it was supposed to be, thank you for bein clear, @shella12 

https://user-images.githubusercontent.com/92229666/226092241-a835371a-27c9-436a-afd4-0f281497fa8f.mp4

shella12 commented on assets/buttonsActions.js:
```diff
@@ -0,0 +1,101 @@
+
+// MENU BUTTONS ACTION
+
+const body = document.querySelector('body');
+const menu = document.querySelector('#menuBar');
+const menuButtons = document.querySelectorAll('.menuAction');
+const hamburguer = document.querySelector('#menuBtn');
+
+menuButtons.forEach((index) => {
```
- [ ] Your hamburger is working great but there is a small issue with it, when menu is displayed and screen resizes the mobile menu disappears but the icons are not toggled. When we resize back to the mobile resolution the `X` icon is displaying in place of `hamburger icon`  and  vice versa. 


https://user-images.githubusercontent.com/44798044/226086256-b2ba7fb3-a44c-4019-90f9-e77b4a987a93.mov

VelzckC0D3 replied:
@shella12 Got it! I've already corrected that too, thanks! 

https://user-images.githubusercontent.com/92229666/226092332-30e0774f-6b4a-4eee-a5d1-10085f74f254.mp4

shella12 reviewed (changes requested):
Hi @VelzckC0D3 , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

<img src="https://media1.giphy.com/media/L422JoNlBBBD42zkQ9/giphy.gif"/>

## Highlights

- Clean code  ✅ 
- Excellent design implementation  ✅ 
- Great UI  ✅ 
- Professional README  ✅ 
- Good commit history  ✅ 
- Descriptive PR  ✅ 

## Suggested changes

- [ ] You have done a great job  👍  making the loom video talking about the features of your website however I would suggest you highlight some interesting piece of code that you have built and are really proud of. As It is also mention in the requirement for the [Project documentation](https://github.com/microverseinc/curriculum-html-css/blob/main/capstone/html_capstone.md#project-documentation):

> Once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a maximum length of 5 minutes. The content of the video should include:
> - A description of the project.
> - A demo of the project features (different pages and different screen sizes).
> - You should also highlight some interesting piece of code or something you built that you are very proud of.


- [ ] _Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

GedeonTS commented on index.html:
```diff
@@ -1,14 +1,254 @@
+      <div id="hero">
+        <h3 class="heroTitle">"hello designers of the world!"</h3>
+        <h1 class="heroTitle">architecture and designers festival <br>2023</h1>
+        <p id="heroDescription">Welcome to the intrnational <span id="heroSpan">Architecture and Design festival
```
- As requested by the previous reviewer [here](https://github.com/VelzckC0D3/capstone_project/pull/1/files#r1140931268)  kindly make sure h1 always come before h2,h3,... in order to meet the semantic HTML syntax.

__Note: only comments starting with [OPTIONAL] are optional, all others are mandatory for the approval of your project__

GedeonTS commented on index.html:
```diff
@@ -1,14 +1,254 @@
+          academics, and activists in the field of architecture and design in a different location every two years.
+          This year, we aim to expand our guest list to include organizations and individuals interested in
+          collaborating with us on joint projects that promote the advancement of the industry, free expression, and
+          open knowledge. If you're actively involved in the world of architecture and design - advocates for
```
- As requested by the previous reviewer kindly make sure you have two HTML files (e.g: index.html and about.html) as per the project requirements :recycle:

VelzckC0D3 replied:
@GedeonTS Oh my god sorry, I didn't commit the last changes but I already did it, it is solved! here it's the 3:36AM I'm a little tired

VelzckC0D3 committed:
Corrections

- Corrected HX hierarchy so it haves correct semantics
- Now, the about page and the home page are 2 separate documents
- Corrected a visual bug on the hamburguer that was shown whenever the resolution was changed and the menu was open

VelzckC0D3 committed:
Merge branch 'website' of https://github.com/VelzckC0D3/capstone_project into website

GedeonTS reviewed (changes requested):
# STATUS: CHANGES REQUIRED :recycle: :
Hi @VelzckC0D3 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

<img src="https://media.giphy.com/media/31Q36EHJ7C5ZoIMw79/giphy.gif">

 ## Highlights :tophat: :+1: 
- No linter errors :+1: 
- Good use of GitHub flow :+1: 
- Good use of media queries :+1: 

 ## Required Changes ♻️
- [OPTIONAL] In my opinion, it would be better to highlight some piece of code that you liked the most in your video as requested by project.
_Check the comments under the review._

## Optional suggestions :ringed_planet: 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

sumairq reviewed (approved):
### Hi @VelzckC0D3 , 

<img src="https://media3.giphy.com/media/BPJmthQ3YRwD6QqcVD/giphy.gif"/>

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Github flow
✔️Descriptive PR


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
